bot/exts/moderation/stream.py

```diff
@@ -0,0 +1,139 @@
+import datetime
+
+import discord
+from async_rediscache import RedisCache
+from discord.ext import commands
+
+from bot.bot import Bot
+from bot.constants import Emojis, Guild, Roles, STAFF_ROLES, VideoPermission
+from bot.converters import Expiry
+from bot.utils.scheduling import Scheduler
+from bot.utils.time import format_infraction_with_duration
+
+
+class Stream(commands.Cog):
+    """Grant and revoke streaming permissions from users."""
+
+    # Stores tasks to remove streaming permission
+    # User id : timestamp relation
+    task_cache = RedisCache()
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+        self.scheduler = Scheduler(self.__class__.__name__)
+        self.reload_task = self.bot.loop.create_task(self._reload_tasks_from_redis())
+
+    async def _remove_streaming_permission(self, schedule_user: discord.Member) -> None:
+        """Remove streaming permission from Member."""
+        await self._delete_from_redis(schedule_user.id)
+        await schedule_user.remove_roles(discord.Object(Roles.video), reason="Streaming access revoked")
```

Comment:
What does Discord do if the permission is revoked while the user is streaming? Does it kick them out or let them keep streaming? What kind of behaviour would we want? @MrHemlock

Reply:
@MarkKoz It does not kick them out.  We would have to boot them from the voice channel.  Currently, I'd prefer if we don't implement anything for it at the moment, as Mods+ do have permissions to boot someone from VC manually.  

Typically if it came down to that, either they'd be receiving a voice ban or some other full infraction, which would bump them off call anyway.  Even a regular server mute does so.

Reply:
In my opinion, the use case for revoke would be if someone finishes up their help session or what have you before it would expire, rather than having to remove it due to bad behavior.